### DISCUSSION TITLE
In lesson  4 when I deployed a basic code for price converstion

### DISCUSSION THREAD

**Discussion by vishvjeet-thakur** [Cyfrin/foundry-full-course-cu#308] [Q&A] [answered]
When I deploy this with 0 ether it is getting deployed successfully but with a non zero ether even with 1 wei it is not getting deployed what could be the reason and insufficient fund is not a reason here.
```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;
import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";

contract FundMe {
     
     uint256 minimumUsd = 5e18;


    function fund() public payable{
          require(priceConversion(msg.value)>minimumUsd, "not enough fund to donate, it should be greater than $5");

    }
    
    function getPrice() public view returns(uint256) {

        AggregatorV3Interface pricefeed = AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306);
        (,int256 answer,,,)= pricefeed.latestRoundData();


        return uint256(answer*1e10);
    }

    function priceConversion(uint256 eth) public view returns(uint256)
    {
        uint256 price  = getPrice();

        return (price*eth)/1e18;

    }
}
```

**Comment by cromewar** [comment]
Hello @vishvjeet-thakur 
This depends a lot of which network you are deploying to and from whom address you are doing so, so please can you give us more context on this?
Please help us to understand:
- the network you are deploying to.
- the wallet you are using to deploy.
- share the script you are using for deployment as well.

**Reply by vishvjeet-thakur** [reply]
Thank you for answering 😊
I am deploying it to the sepolia test network on metamask .
Code for the contract
' ' '
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;
import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";

contract FundMe {
     
     uint256 minimumUsd = 5e18;


    function fund() public payable{
          require(priceConversion(msg.value)>minimumUsd, "not enough fund to donate, it should be greater than $5");

    }
    
    function getPrice() public view returns(uint256) {

        AggregatorV3Interface pricefeed = AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306);
        (,int256 answer,,,)= pricefeed.latestRoundData();


        return uint256(answer*1e10);
    }

    function priceConversion(uint256 eth) public view returns(uint256)
    {
        uint256 price  = getPrice();

        return (price*eth)/1e18;

    }
}
' ' '


error that I am getting 
' ' ' 
Gas estimation errored with the following message (see below). The transaction execution will likely fail. Do you want to force sending?
Returned error: {"jsonrpc":"2.0","error":"execution reverted","id":7363640243267103}
' ' '

**Reply by DevSwayam** [reply]
There is syntax error of your require statement in your fund() function. 
It should be ```function fund() public payable{
          require(msg.value.priceConversion()>minimumUsd, "not enough fund to donate, it should be greater than $5");
    }```
instead of ``` function fund() public payable{
          require(priceConversion(msg.value)>minimumUsd, "not enough fund to donate, it should be greater than $5");
 }```

**Reply by vishvjeet-thakur** [reply]
@DevSwayam  bro  but priceCoversion is a function I externally made and using it as a method of msg.value is giving a error

**Reply by DevSwayam** [reply]
You can ask chatgpt on why provided syntax is correct at least it is working for my case.

**Reply by DevSwayam** [reply]
![Screenshot_2023-07-11-09-01-56-30_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/Cyfrin/foundry-full-course-f23/assets/99035115/faadf278-c631-4f98-b007-d1d036605514)
You can always take help of your AI buddy

**Reply by vishvjeet-thakur** [reply, marked as answer]
Thankyou guys @cromewar @DevSwayam  just because of you guys learning something new is very easy.
Actually after looking again it turned out that I was doing a mistake in deploying like I was unaware that we have to fill value after deployment but I was doing it before that so that why it was giving that error.
But thank alot again.
